recipes/cwinpy/meta.yaml

```
@@ -0,0 +1,62 @@
+{% set name = "cwinpy" %}
+{% set version = "0.2.1" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 3caedf3aac6b8ef9bc0df8467e1610587c1fae3b2a0266ee6c839d45a79e5cb0
+
+build:
+  entry_points:
+    - cwinpy_knope = cwinpy.knope.knope:knope_cli
+    - cwinpy_knope_dag = cwinpy.knope.knope:knope_dag_cli
+    - cwinpy_knope_generate_pp_plots = cwinpy.knope.testing:generate_pp_plots
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+    - setuptools
+  run:
+    - astropy
+    - bilby >=0.5.8
+    - bilby_pipe >=0.2.0
+    - gwpy
+    - lintegrate
+    - matplotlib
```

Only install base if you don't require the Qt backend
```suggestion
    - matplotlib-base
```

---

@duncanmmacleod the code primarily uses matplotlib via gwpy. I see that gwpy's recipe uses `matplotlib`, so do you have an opinion on this?

---

if your library doesn't depend upon interactive images, but rather uses `matplotlib` to render images to files, then just depend upon `matplotlib-base` as suggested.

---

since `gwpy` depends upon `matplotlib` in full, this won't really make a difference to installing your library anyway, but I might update the `gwpy` recipe in the near future.

---

https://github.com/conda-forge/gwpy-feedstock/pull/17